coriolisclient/tests/v1/test_logging.py

```diff
@@ -0,0 +1,278 @@
+            "query_args": {
+                "arg1": None,
+                "arg2": None
+            },
+            "is_websocket": True,
+            "expected_result":
+                "ws:///None/sentinel.resource"
+        },
+        {
+            "query_args": {
+                "arg1": None,
+                "arg2": "mock_arg2"
+            },
+            "_ep_url": "https:///ep_url",
+            "is_websocket": True,
+            "expected_result":
+                "wss:///ep_url/sentinel.resource?arg2=mock_arg2"
+        },
+        {
+            "query_args": {
+                "arg1": "mock_arg1",
+                "arg2": "mock_arg2"
+            },
+            "_ep_url": "https:///ep_url",
+            "is_websocket": False,
+            "expected_result": "https:///ep_url/sentinel.resource"
+                "?arg1=mock_arg1&arg2=mock_arg2"
+        }
+    )
+    @mock.patch.object(logging.LoggingClient, '_get_endpoint_url')
+    def test_construct_url(self, data, mock_get_endpoint_url):
+        self.logger._ep_url = None
+        mock_get_endpoint_url.return_value = data.get("_ep_url", None)
+
+        result = self.logger._construct_url(
+            mock.sentinel.resource,
+            data.get("query_args"),
+            is_websocket=data.get("is_websocket", False),
+        )
+
+        self.assertEqual(
+            data.get("expected_result"),
+            result
+        )
+
+    @ddt.data(
+        (None, None, False, False),
+        ("1", 1, False, False),
+        ("1234567890123456789", None, True, True),
+        ("abc", None, True, True),
+        ("", None, True, True),
+        ("10s", mock.ANY, False, True),
+    )
+    @ddt.unpack
+    def test_convert_period_to_timestamp(
+        self,
+        period,
+        expected_result,
+        raises,
+        has_logs
+    ):
+        if raises is False:
+            if has_logs is False:
+                result = self.logger._convert_period_to_timestamp(period)
+            else:
+                with self.assertLogs(logger=logging.LOG, level="WARNING"):
+                    result = self.logger._convert_period_to_timestamp(period)
+            self.assertEqual(
+                expected_result,
+                result
+            )
+        else:
+            with self.assertLogs(logger=logging.LOG, level="WARNING"):
+                self.assertRaises(
+                    exceptions.CoriolisException,
+                    self.logger._convert_period_to_timestamp,
+                    period
+                )
+
+    @mock.patch.object(requests, "get")
+    @mock.patch.object(logging.LoggingClient, "_construct_url")
+    @mock.patch.object(logging.LoggingClient, "_convert_period_to_timestamp")
+    def test_download_logs(
+        self,
+        mock_convert_period_to_timestamp,
+        mock_construct_url,
+        mock_get
+    ):
+        logs_path = os.path.dirname(os.path.realpath(__file__))
+        logs_path = os.path.join(logs_path, 'data')
```

Comment:
Please setup a tempfile instead of pushing test results to repository :) https://docs.python.org/3/library/tempfile.html